Client/mods/deathmatch/logic/CClientModel.cpp

```diff
@@ -53,20 +63,45 @@ bool CClientModel::Deallocate(void)
     // ModelInfo must be valid
     if (pModelInfo->IsValid())
     {
-        if (m_eModelType == CCLIENTMODELPED)
+        if (m_eModelType == eClientModelType::PED)
         {
             // If some ped is using this ID, change him to CJ
             CClientPedManager*                       pPedManager = g_pClientGame->GetManager()->GetPedManager();
-            std::vector<CClientPed*>::const_iterator iter = pPedManager->IterBegin();
-            for (; iter != pPedManager->IterEnd(); iter++)
+            for (auto iter = pPedManager->IterBegin(); iter != pPedManager->IterEnd(); iter++)
+            {
+                CClientPed* pPed = *iter;
+                if (pPed->GetModel() == m_iModelID)
+                {
+                    if (pPed->IsStreamedIn())
+                    {
+                        pPed->StreamOutForABit();
+                    }
+                    pPed->SetModel(0);
+
+                    CLuaArguments Arguments;
+                    Arguments.PushNumber(m_iModelID);
+                    Arguments.PushNumber(0);
+                    pPed->CallEvent("onClientElementModelChange", Arguments, true);
+                }
+            }
+        }
```

Comment:
Shorter methods are very easy to read. You can split this code into a function:
```
void CClientPedManager::triggerElementModelChangeEvent(unsigned short modelId)
{
    for (auto& ped : m_List)
    {
        if (ped->GetModel() == modelId)
        {
            if (ped->IsStreamedIn())
                ped->StreamOutForABit();
            ped->SetModel(0);
            CLuaArguments Arguments;
            Arguments.PushNumber(modelId);
            Arguments.PushNumber(0);
            ped->CallEvent("onClientElementModelChange", Arguments, true);
        }
    }
}
```
```
g_pClientGame->GetManager()->GetPedManager()->triggerElementModelChangeEvent(m_iModelID);
```

Reply:
also from another [PR](https://github.com/multitheftauto/mtasa-blue/pull/1701/commits/efb87d85221f90f78aafc60c6c09d564ec887dca#diff-73eeec935449ae07c6300011119ba1692ac7448a961d7a2498abe8ed142e4343R77)